lib/test/karma.js

```diff
@@ -16,7 +18,69 @@ function quote(value) {
   return '"' + value + '"';
 }
 
+var defaultReporterName = 'angularity-karma-reporter';
+
 var filesAppendRegex = /\/\*+\s*ANGULARITY_FILE_LIST\s*\*+\// ;
+var reportersAppendRegex = /\/\*+\s*ANGULARITY_REPORTER_LIST\s*\*+\// ;
+var pluginsAppendRegex = /\/\*+\s*ANGULARITY_PLUGIN_LIST\s*\*+\// ;
+var karmaReporterMatchNameRegex = /^karma-(.+)-reporter$/ ;
+
+/**
+ * Determine the `require` path for a plugin.
+ *
+ * @param {string} reporterName Either a package name,
+ *   *OR* an absolute path, for a karma reporter.
+ * @return {string} In karma config file, `plugins` will be inline `require`d
+ *   using this return value.
+ */
+function getKarmaReporterPluginPath(reporterName) {
+  if (typeof reporterName !== 'string') {
+    throw 'Get Karma Reporter Plugin Path: Reporter name is unspecified';
+  }
+
+  if (reporterName === defaultReporterName) {
+    return defaultReporterName;
+  }
+  else {
+    var reporterPath = (path.dirname(reporterName) === '.') ?
+      path.resolve('node_modules', 'karma-'+reporterName+'-reporter') :
+      reporterName;
+    try {
+      require(reporterPath);
+    }
+    catch (ex) {
+      throw 'Get Karma Reporter Plugin Path: Attempt to require reporter from path '+reporterPath+' with no success.';
```

Comment:
These throws are intended to be caught and printed as strings within `checkKarmaReporter.
Overhead of capturing and storing the stack trace is unnecessary here.